src/_roots.jl

```diff
@@ -39,10 +39,11 @@ end
 closeenough(a::T, b::T, ϵ) where {T<:AbstractFloat} = isapprox(a, b; rtol=0, atol=ϵ)
 closeenough(a::T, b::T, _) where {T<:Integer} = a == b
 
-function refine_grid(grid, ::Val{α}) where {α}
+function refine_grid(grid::Vector{T}, ::Val{α}) where {T, α}
+    isempty(grid) && return grid
     n = length(grid)
     newn = α * (n - 1) + 1
-    newgrid = Vector{eltype(grid)}(undef, newn)
+    newgrid = Vector{float(T)}(undef, newn)
```

Comment:
This change is required because the `newn` gets a negative value for empty input, i.e., `grid=Float64[]`; otherwise, `refine_grid([])` will crash a Julia process.

```
julia> using SparseIR

julia> SparseIR.refine_grid(Float64[], Val(2))
ERROR: ArgumentError: invalid GenericMemory size: too large for system address width
Stacktrace:
 [1] GenericMemory
   @ ./boot.jl:516 [inlined]
 [2] Array
   @ ./boot.jl:578 [inlined]
 [3] refine_grid(grid::Vector{Float64}, ::Val{2})
   @ SparseIR ~/.julia/packages/SparseIR/iqwia/src/_roots.jl:45
 [4] top-level scope
   @ REPL[2]:1
```

Reply:
The current implementation (on the main branch) causes an error.

```julia
julia> SparseIR.refine_grid([1,2,3], Val(2))
ERROR: InexactError: Int64(1.5)
Stacktrace:
 [1] Int64
   @ ./float.jl:994 [inlined]
 [2] convert
   @ ./number.jl:7 [inlined]
 [3] setindex!
   @ ./array.jl:987 [inlined]
 [4] refine_grid(grid::Vector{Int64}, ::Val{2})
   @ SparseIR ~/.julia/packages/SparseIR/iqwia/src/_roots.jl:53
 [5] top-level scope
   @ REPL[3]:1
```

To avoid this issue, `newgrid = Vector{float(T)}(undef, newn)` is required.